assert/assertions_test.go

```diff
@@ -2845,52 +2845,202 @@ func Test_truncatingFormat(t *testing.T) {
 	}
 }
 
+func parseLabeledOutput(output string) []labeledContent {
```

Comment:
Why is this complex processing necessary? I don't get why you are not just matching the raw message.

At least a comment should explain what this does and why. Examples would be welcome.

Reply:
I added a comment describing what this function does. It's basically the inverse of `labeledOutput()`.

> Why is this complex processing necessary? I don't get why you are not just matching the raw message.

My goal is to  want to capture the error message passed to `assert.Fail()`. But I can't do that directly; instead, I need to use `captureTestingT` to capture output written to `t.Errorf()`. This output is formatted from a number of labels. So a natural first step is to process this output back into the original set of `labeledContent` that produced it.

Once I have the original `labeledContent`, I can match the error message against the content for the **Error** label.

Read `assert.Fail()` to see what output it passes to `t.Errorf`.

Reply:
`parseLabeledOutput` would also be useful for other things in the future, like testing other labeled content, while ignoring labels like **Error Trace**. It could also be used to test `labeledOutput` itself if desired.